block/batch.go

```diff
@@ -6,14 +6,16 @@ import (
 
 // BatchQueue is a queue of transaction batches with timestamps
 type BatchQueue struct {
-	queue []BatchWithTime
-	mu    sync.Mutex
+	queue    []BatchWithTime
+	mu       sync.Mutex
+	notifyCh chan struct{}
 }
 
 // NewBatchQueue creates a new BatchQueue
 func NewBatchQueue() *BatchQueue {
 	return &BatchQueue{
-		queue: make([]BatchWithTime, 0),
+		queue:    make([]BatchWithTime, 0),
+		notifyCh: make(chan struct{}, 1), // Buffered channel to avoid blocking
```

Comment:
The buffer doesn't prevent blocking. You still need to use select statements. 
An empty channel is just useless because you can never put anything into it. 

https://go.dev/play/p/wPvDbnaX983

I'd remove the comment as it is inaccurate.